From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Thu, 4 Oct 2018 14:22:09 -0400
Subject: [PATCH] Reimplement boot_counter

This adds "increment" and "decrement" commands, and uses them to maintain our
variables in 01_fallback_counter.  It also simplifies the counter logic, so
that there are no nested tests that conflict with each other.

Apparently, this *really* wasn't tested well enough.

Resolves: rhbz#1614637
Signed-off-by: Peter Jones <pjones@redhat.com>
---
 util/grub.d/01_fallback_counting.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/util/grub.d/01_fallback_counting.in b/util/grub.d/01_fallback_counting.in
index 6ca13da03df..1a0b5fbecfc 100644
--- a/util/grub.d/01_fallback_counting.in
+++ b/util/grub.d/01_fallback_counting.in
@@ -1,5 +1,9 @@
 #! /bin/sh -e
 
+if ! systemctl -q is-enabled greenboot.service >/dev/null 2>&1 ; then
+  exit 0
+fi
+
 # Boot Counting
 cat << EOF
 insmod increment
